### PR TITLE
feat: elements naming

### LINKED ISSUE #8: Elements names instead of element types
Feature to name/rename elements by double clicking on them in element's tab. I think it'll be very useful in cases, when there will be a lot of boxes on the playground.

### PR DESCRIPTION
**Pull request by vshakitskiy**
+ close #8

### CODE REVIEW

File: apps/dashboard/src/components/RightPanel/NameSection.tsx
Comment (QuiiBz) [inline apps/dashboard/src/components/RightPanel/NameSection.tsx:1]: Not sure about this new section, it feels a bit redundant.

Screenshot 2023-12-30 at 14 22 05

What do you think of removing it, and only keep the double-click on the left panel?

File: apps/dashboard/src/components/LeftPanel/ElementRow.tsx [hunk 1]
```diff
@@ -29,11 +32,47 @@ export function ElementRow({ element }: ElementRowProps) {
...
     <div className="flex justify-between items-center pb-2 cursor-auto" ref={setNodeRef} style={style} {...attributes} {...listeners}>
       <button
-        className={`flex gap-2 select-none text-gray-600 hover:text-gray-900 w-full ${selectedElement === element.id ? '!text-blue-500' : ''} ${!element.visible ? '!text-gray-300' : ''}`}
+        className={`flex items-center gap-2 select-none text-gray-600 hover:text-gray-900 w-full ${selectedElement === element.id ? '!text-blue-500' : ''} ${!element.visible ? '!text-gray-300' : ''}`}
         onClick={() => { setSelectedElement(element.id); }}
+        onDoubleClick={() => {
+          if (isEditing)
+            return
+
+          setIsEditing(true)
```
Comment (QuiiBz) [inline apps/dashboard/src/components/LeftPanel/ElementRow.tsx:74]: There seem to be an issue where we don't focus the input right after double-clicking, we means we have to double-click + click again to actually start editing the name:

https://github.com/QuiiBz/ogstudio/assets/43268759/784e4ed4-969f-4378-b715-0d4ca9bc3fd9

File: apps/dashboard/src/components/LeftPanel/ElementRow.tsx [hunk 2]
```diff
@@ -56,7 +95,35 @@ export function ElementRow({ element }: ElementRowProps) {
...
+            <button
+              className="text-gray-600 hover:text-gray-900"
+              type="submit"
+            >
+              <CheckIcon />
+            </button>
+          </form>
+        ) : (<>
+          {element.name}
+        </>)}
```
Comment (QuiiBz) [inline apps/dashboard/src/components/LeftPanel/ElementRow.tsx:126]: I *think* we can refactor this a bit:

```suggestion
        ) : element.name}
```

**Review by QuiiBz — CHANGES_REQUESTED** [review]
That's gonna be useful! Some comments:

File: apps/dashboard/src/components/RightPanel/NameSection.tsx
Comment (vshakitskiy) [reply apps/dashboard/src/components/RightPanel/NameSection.tsx:1]: Let's keep only on left panel then.

File: apps/dashboard/src/components/LeftPanel/ElementRow.tsx (hunk 1 above)
Comment (vshakitskiy) [reply apps/dashboard/src/components/LeftPanel/ElementRow.tsx:74]: I think setting autofocus on inputs will solve the issue.

**Comment by QuiiBz** [comment]
Could you rebase the branch? I made some changes in #27 around the file structure. Then we'll be ready to merge!

**Pull request closed without merging**